### ISSUE TITLE
roles: Create endpoints for role management

### ISSUE THREAD

**Issue by BrunoRosendo** [NIAEFEUP/website-niaefeup-backend-old#79]
After the work done in #49, where the roles and permissions were created and associated with accounts and activities, we now need to use them to implement our role management system.

Note that managing access from permissions is not part of this issue, which should be done in or after #60. That said, this issue should tackle (more or less) the following tasks:

- [ ] Create controllers for the following features (edit to include more if needed):
  - [ ] Create and delete roles associated with a set of permissions. Watch out for the activity permissions too (e.g. `PerActivityRole` model and `getEffectivePermissionsForActivity` function in `Account` model)
  - [ ] Add/Revoke permissions to/from roles, including activity roles. One can also consider a similar implementation to the "create" endpoint, identical to other controllers where entities are "recreated".
  - [ ] Add/Revoke roles to/from accounts. "Recreating" an account is not a good idea here
- [ ] Make sure to create integration tests (and unit if necessary) for the endpoints you create. Since #49 didn't implement any controllers, only unit tests are present, meaning that testing the roles' functionalities is a very important part of this issue.
- [ ] Watch out for the server's responses. There might be cases where role information is unexpectedly returned and cases where something important is missing. This is due to the choice between lazy and eager loading when serializing the models (check discussion in https://github.com/NIAEFEUP/website-niaefeup-backend/pull/49#issuecomment-1376139238). Update fetch type when necessary
